decidim-core/lib/decidim/maintenance/taxonomy_importer.rb

```diff
@@ -0,0 +1,132 @@
+
+        name = taxonomy.name[organization.default_locale]
+
+        begin
+          if resource.respond_to? :taxonomy
+            resource.update(decidim_taxonomy_id: taxonomy.id)
+          else
+            return if resource.taxonomies.include?(taxonomy)
+
+            resource.taxonomies << taxonomy
+          end
+          result[:taxonomies_assigned][name] ||= []
+          result[:taxonomies_assigned][name] << object_id unless result[:taxonomies_assigned][name].include?(object_id)
+        rescue ActiveRecord::RecordInvalid
+          result[:failed_resources] << object_id
+        end
+      end
+
+      def import_filter(root, data)
+        filter = find_taxonomy_filter!(root, data)
+
+        data["items"].each do |item_names|
+          taxonomy = root
+          item_names.each do |item_name|
+            taxonomy = find_taxonomy!(taxonomy.children, item_name)
+          end
+          next if filter.filter_items.exists?(taxonomy_item: taxonomy)
+
+          filter.filter_items.create!(taxonomy_item: taxonomy) do
+            result[:filters_created]["#{filter.space_manifest}: #{filter.internal_name[I18n.locale.to_s]}"] ||= []
+            result[:filters_created]["#{filter.space_manifest}: #{filter.internal_name[I18n.locale.to_s]}"] << item_names.join(" > ")
+          end
+        end
+
+        data["components"].each do |component_id|
+          component = GlobalID::Locator.locate(component_id)
+          if component
+            begin
+              component.update!(settings: { taxonomy_filters: [filter.id.to_s] })
+              result[:components_assigned]["#{filter.space_manifest}: #{filter.internal_name[I18n.locale.to_s]}"] ||= []
+              result[:components_assigned]["#{filter.space_manifest}: #{filter.internal_name[I18n.locale.to_s]}"] << component_id
+            rescue ActiveRecord::RecordInvalid
+              result[:failed_components] << component_id
+            end
+          else
+            result[:failed_components] << component_id
+          end
+        end
+      end
+
+      def find_taxonomy(association, name)
+        association.find_by("name->>? = ?", organization.default_locale, name)
+      end
+
+      def find_taxonomy!(association, name)
+        find_taxonomy(association, name) || create_taxonomy!(association, name)
+      end
+
+      def create_taxonomy!(association, name)
+        association.create!(name: { organization.default_locale => name }, organization:) do
+          result[:taxonomies_created] << name
+        end
+      end
+
+      def find_taxonomy_filter!(root_taxonomy, data)
+        name = data["internal_name"] || data["name"]
+        attributes = {
+          space_filter: data["space_filter"],
+          space_manifest: data["space_manifest"]
+        }
+        attributes["internal_name"] = { I18n.locale.to_s => data["internal_name"] } if data["internal_name"]
+        attributes["name"] = { I18n.locale.to_s => data["public_name"] } if data["public_name"]
```

Comment:
maybe is worth looking for organization locale ?
```suggestion
        attributes["internal_name"] = {organization.default_locale.to_s => data["internal_name"] } if data["internal_name"]
        attributes["name"] = { organization.default_locale.to_s => data["public_name"] } if data["public_name"]
```